authenticators/components/org.wso2.carbon.identity.sample.local.authenticator/src/main/java/org/wso2/carbon/identity/sample/local/authenticator/SampleLocalAuthenticator.java

```diff
@@ -0,0 +1,169 @@
+
+        Optional<org.wso2.carbon.user.core.common.User> user = Optional.empty();
+
+        boolean isAuthenticated = false;
+
+        // Check the authentication.
+        try {
+            int tenantId = IdentityTenantUtil.getTenantIdOfUser(username);
+            UserRealm userRealm = SampleLocalAuthenticatorServiceComponent.getRealmService()
+                    .getTenantUserRealm(tenantId);
+            if (userRealm != null) {
+                UniqueIDUserStoreManager userStoreManager = (UniqueIDUserStoreManager) userRealm.getUserStoreManager();
+
+                // This custom local authenticator is using the telephone number as the username.
+                // Therefore the login identifier claim is http://wso2.org/claims/telephone.
+                AuthenticationResult authenticationResult = userStoreManager.
+                        authenticateWithID(MOBILE_CLAIM_URL, username, password, UserCoreConstants.DEFAULT_PROFILE);
+                if (AuthenticationResult.AuthenticationStatus.SUCCESS == authenticationResult.getAuthenticationStatus()) {
+                    user = authenticationResult.getAuthenticatedUser();
+                    isAuthenticated = true;
+                }
+            } else {
+                if (log.isDebugEnabled()) {
+                    log.debug("Custom authentication failed since the user realm for the given tenant, " +
+                            tenantId + " is null.");
+                }
+                throw new AuthenticationFailedException("Cannot find the user realm for the given tenant: " + tenantId,
+                        User.getUserFromUserName(username));
+            }
+        } catch (IdentityRuntimeException e) {
+            if (log.isDebugEnabled()) {
+                log.debug("Custom authentication failed while trying to get the tenant ID of the user " + username, e);
+            }
+            throw new AuthenticationFailedException(e.getMessage(), e);
+        } catch (org.wso2.carbon.user.api.UserStoreException e) {
+            if (log.isDebugEnabled()) {
+                log.debug("Custom authentication failed while trying to authenticate the user " + username, e);
+            }
+            throw new AuthenticationFailedException(e.getMessage(), e);
+        }
+
+        // If the authentication fails, throws the invalid client credential exception.
+        if (!isAuthenticated) {
+            if (log.isDebugEnabled()) {
+                log.debug("User authentication failed due to invalid credentials");
+            }
+            throw new InvalidCredentialsException("User authentication failed due to invalid credentials",
+                    User.getUserFromUserName(username));
+        }
+
+        // When the user is successfully authenticated, add the user to the authentication context to be used later in
+        // the process.
+        if (user != null) {
+            username = user.get().getUsername();
```

Comment:
is the `username` here different from the value of `httpServletRequest.getParameter(USERNAME)`?